localcache/impl.go

```diff
@@ -0,0 +1,64 @@
+package localcache
+
+import (
+	"sync"
+	"time"
+)
+
+const expireTime = 30 * time.Second
+
+type clock interface {
+	Now() time.Time
+}
+
+type realClock struct{}
+
+func (realClock) Now() time.Time {
+	return time.Now()
+}
+
+type cache struct {
+	data   map[string]interface{}
+	expiry map[string]time.Time
+	mutex  sync.RWMutex
+	clock  clock
+}
+
+func New(c clock) Cache {
+	if c == nil {
+		c = realClock{}
+	}
+	return &cache{
+		data:   make(map[string]interface{}),
+		expiry: make(map[string]time.Time),
+		clock:  c,
+	}
+}
+
+// Get returns the value for the given key, or nil if the key is not found or expired
+func (c *cache) Get(key string) interface{} {
```

Comment:
如果有問題可能會偏好回傳獨立的錯誤，這邊用 nil 把取得結果跟錯誤放在一起用比較模糊一點
```suggestion
func (c *cache) Get(key string) (interface{}, error) {
```